### ISSUE TITLE
Comments in Lists

### ISSUE THREAD

**Issue by Foggalong** [markdownlint/markdownlint#390]
## Description

If there are comments inside lists the linter gets unhappy that that there isn't adequate spacing around lists.

### Environment

Tested with `mdl` v0.11.0 on Linux. Same issue is also present in derived linters, e.g. DavidAnson/markdownlint#431

### Expected Behavior

markdownlint ignores MD032 when the line next to the list is a comment, possibly verifying whether the next line on is a space or another list item.

### Actual Behavior

MD032 rule shows as broken.

## Replication Case

```markdown
- item <!-- comment -->
- item <!-- comment -->
<!--
  - commented subitem: description
  - commented subitem: description
-->
- item <!-- comment -->
- item <!-- comment -->
```

Markdown lint is complaining that lines 2 and 7 are MD032 violations because there aren't spaces around them, but if spaces are added (between 2 & 3 and 6 & 7) then that separates the sublist from the list and, when uncommented, leads to MD012 being flagged. Alternatively, the comment can be adjusted so that it looks like

```markdown
- item <!-- comment -->
- item <!-- comment -->
<!--- commented subitem: description
    - commented subitem: description-->
- item <!-- comment -->
- item <!-- comment -->
```

but (as well as looking weird) that flags a MD007 error when the sublist is uncommented.

From what I can tell there isn't a way to structure comments in lists without breaking one of those three rules. The context for this issue was [this GitHub issue template](https://github.com/jdm-contrib/jdm/blob/master/.github/ISSUE_TEMPLATE/new_entry.md) which has a [commented sublist](https://raw.githubusercontent.com/jdm-contrib/jdm/master/.github/ISSUE_TEMPLATE/new_entry.md) and so seemingly can never be markdownlint compliant.
